src/main/java/mate/academy/spring/controller/CustomGlobalExceptionHandler.java

```diff
@@ -0,0 +1,54 @@
+package mate.academy.spring.controller;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import mate.academy.spring.exception.DataProcessingException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+    @Override
+    protected ResponseEntity<Object>
+            handleMethodArgumentNotValid(MethodArgumentNotValidException ex,
+                                                                  HttpHeaders headers,
+                                                                  HttpStatus status,
+                                                                  WebRequest request) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("status", status.value());
+        List<String> errors = ex.getBindingResult()
+                .getAllErrors()
+                .stream()
+                .map(this::getErrorMessage)
+                .collect(Collectors.toList());
+        body.put("errors", errors);
+        return new ResponseEntity<>(body, headers, status);
+    }
+
+    @ExceptionHandler(DataProcessingException.class)
+    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
+    @ResponseBody
+    public Object processDataValidationException(DataProcessingException ex) {
```

Comment:
Let's return JSON object with fields like `status`, `message`, and `exceptionName` instead of simple string
```suggestion
    public ResponseEntity<Object> handleDataProcessingException(DataProcessingException ex) {
```